docker/templates/python/{{cookiecutter.directory_name}}/docker-compose.yml

```diff
@@ -0,0 +1,68 @@
+version: '2.4'
+
+services:
+  app:
+    image: {{cookiecutter.app_name}}
+    container_name: {{cookiecutter.app_name}}
+    restart: always
+    build: .
+    # Allow container to be attached to, e.g., to access the pdb shell
+    stdin_open: true
+    tty: true
+    ports:
+      # Map ports on your computer to ports on your container. This allows you,
+      # e.g., to visit your containerized application in a browser on your
+      # computer.
+      - 8000:8000
+    depends_on:
+      postgres:
+        condition: service_healthy
+    volumes:
+      # Mount the development directory as a volume into the container, so
+      # Docker automatically recognizes your changes.
+      - .:/app{% if cookiecutter.local_settings != 'None' %}
+      - ${PWD}/{{cookiecutter.local_settings}}:/app/{{cookiecutter.local_settings|replace(".example", "")}}{% endif %}
+    command: {{cookiecutter.run_command}}{% if cookiecutter.auto_migrate == 'True' %}
+
+  migration:
+    container_name: {{cookiecutter.app_name}}-migration
+    image: {{cookiecutter.app_name}}:latest
+    depends_on:
+      # Declaring this dependency ensures that your application image is built
+      # before migrations are run, and that your application and migrations can
+      # be run from the same image, rather than creating purpose-specific
+      # copies.
+      - app
+    volumes:
+      # These should generally be the same as your application volumes.
+      - .:/app{% if cookiecutter.local_settings != 'None' %}
+      - ${PWD}/{{cookiecutter.local_settings}}:/app/{{cookiecutter.local_settings|replace(".example", "")}}{% endif %}
+    command: {{cookiecutter.migrate_command}}
+{% endif %}
+  postgres:
+    container_name: {{cookiecutter.app_name}}-postgres
+    image: {% if cookiecutter.postgis == 'True' %}mdillon/postgis{% else %}postgres{% endif %}:{{cookiecutter.pg_version}}
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready -U postgres"]
+      interval: 10s
+      timeout: 5s
+      retries: 5
+    environment:
+      # The default Postgres image exposes a number of environmental variables
+      # that allow you to configure the container's behavior, without writing
+      # any additional code. Specify the name of your database, and any other
+      # variables, here. https://hub.docker.com/_/postgres/#environment-variables
+      - POSTGRES_DB={{cookiecutter.pg_db}}{% if cookiecutter.postgis == 'True' %}
+      - POSTGRESS_PASSWORD='postgres'{% endif %}
```

Comment:
Does the Postgis image require a `POSTGRES_PASSWORD` variable? (And should it be spelled `POSTGRESS`?)

Reply:
i thought so, but i'll test it without one. also, no, it should not be spelled `POSTGRESS` 😂

Reply:
update: using this postgis image without a password will raise `django.db.utils.OperationalError: fe_sendauth: no password supplied`. 

it's a little weird to me that this is different than using the regular postgres image, because [the docs for the regular postgres image](https://hub.docker.com/_/postgres/) suggest it, too, should require a password:

> Note 1: The PostgreSQL image sets up trust authentication locally so you may notice a password is not required when connecting from localhost (inside the same container). However, a password will be required if connecting from a different host/container.

in any case, these templates account for the discrepancy. ok by you?